src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.cs

```diff
@@ -43,5 +43,68 @@ public sealed override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationAction(CompilationAnalyzer.AnalyzeCompilation);
         }
+
+        protected bool IsNotSupportedByLiveAnalysis(int code)
+        {
+            if (code == MessageProvider.ERR_FailedToCreateTempFile ||
```

Comment:
@mavasani  but compiler does produce those errors through csc/vbc. just CompilerAnalyzer never produce them.

I think you are treating compiler and compiler analyzer same. but that is not true. compiler is superset of compiler analyzer. and current implementation of SupportedDiagnostics returns diagnostic descriptors for compiler rather than compileranalyzer. and that is where all problem comes in.